fake_test.go

```diff
@@ -558,3 +558,180 @@ func TestFakeAddInsertReplace(t *testing.T) {
 
 	assertRules(t, fake, "thirteenth", "sixth", "twelfth", "fifth", "seventh", "ninth", "eighth", "fourth", "third", "eleventh", "tenth")
 }
+
+func TestFakeParseDump(t *testing.T) {
+
+	tc1 := strings.TrimPrefix(dedent.Dedent(`
+		add table ip kube-proxy
+		add chain ip kube-proxy anotherchain
+		add chain ip kube-proxy chain { comment "foo" ; }
+		add map ip kube-proxy map1 { type ipv4_addr . inet_proto . inet_service ; }
+		add rule ip kube-proxy anotherchain ip saddr 1.2.3.4 drop comment "drop rule"
+		add rule ip kube-proxy anotherchain ip daddr 5.6.7.8 reject comment "reject rule"
+		add rule ip kube-proxy chain ip daddr 10.0.0.0/8 drop
+		add rule ip kube-proxy chain masquerade comment "comment"
+		add element ip kube-proxy map1 { 192.168.0.1 . tcp . 80 : drop }
+		add element ip kube-proxy map1 { 192.168.0.2 . tcp . 443 comment "with a comment" : goto anotherchain }
+		`), "\n")
+
+	tc2 := dedent.Dedent(`
```

Comment:
Is it intentional that you use TrimPrefix on tc1 but not tc2?

You could also do the TrimPrefix'ing and Dedent'ing in the loop below rather than inline here